src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -313,7 +314,10 @@ private string FinishFixingName(string name)
             {
                 words = name.Split(new[] { WordSeparator }, StringSplitOptions.RemoveEmptyEntries);
             }
-
+            if(CapitalizationScheme == Capitalization.AllLower && !string.IsNullOrEmpty(WordSeparator))
+            {
+                words = Regex.Split(name, @"(?<!^)(?=[A-Z])");
```

Comment:
I believe we already have code in Roslyn that splits identifiers into "words"

Reply:
@dotnet/roslyn-ide  : I'm not finding that API. Does anyone know which I'm thinking of?

Reply:
Also, @CyrusNajmabadi

Reply:
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Shared/Utilities/StringBreaker.cs,e4e6edc6966e2d2f